### PR TITLE
Add Xcode environment support

### PR DESCRIPTION
**Pull request by rwalker-apple**
#### Problem
 It is necessary to set arbitrary environment variables for Xcode builds in some environments, but chip_xcode_build_connector.sh only supports one or two.

 #### Summary of Changes
 * add updirs(), which looks for a file in parent directories
 * use updirs() to find and source an _env.sh script for the connector

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@rwalker-apple Are the temperature-measurement bits here on purpose?

**Comment by rwalker-apple** [comment]
> @rwalker-apple Are the temperature-measurement bits here on purpose?

no, I think I have a borken master ;) fixed now

**Comment by rwalker-apple** [comment]
@mspang @andy31415  @jelderton @BroderickCarlin @hawk248 ?

**Comment by rwalker-apple** [comment]
@bzbarsky-apple ?

File: scripts/bootstrap.sh [hunk 1]
```diff
@@ -14,10 +14,18 @@
 # limitations under the License.
 #
 
+if [[ ! -d ${CHIP_ROOT} ]]; then
```
Comment (bzbarsky-apple) [inline scripts/bootstrap.sh:17]: This seems like a mis-merge with <https://github.com/project-chip/connectedhomeip/pull/3291>?  Or at least looks like it backs out the changes from that?

File: src/darwin/Framework/chip_xcode_build_connector.sh [hunk 2]
```diff
@@ -100,14 +100,35 @@ fi
     )
 }
 
+updirs() {
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/chip_xcode_build_connector.sh:103]: Maybe `find_in_ancestors` to be clearer?

File: src/darwin/Framework/chip_xcode_build_connector.sh [hunk 3]
```diff
@@ -100,14 +100,35 @@ fi
...
+    declare lookinfer="${1}"
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/chip_xcode_build_connector.sh:104]: That's meant to be "looking for"?  It took me a while to figure out the naming....  Maybe call it `to_find` or some such?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/darwin/Framework/chip_xcode_build_connector.sh (hunk 3 above)
Comment (rwalker-apple) [reply src/darwin/Framework/chip_xcode_build_connector.sh:104]: yes, "looking for"

I wrote this code years ago for my own dotfiles

**Review by mspang — APPROVED** [review]

**Pull request merged**